pcdsdevices/epics_motor.py

```diff
@@ -1038,6 +1038,26 @@ def home(self, direction=None, wait=True, **kwargs):
 
         return status
 
+    def check_value(self, pos: float) -> None:
+        """
+        Check that the motor can reach the position.
+
+        Inherits limits and related checks from parent classes.
+        Adds one additional check: the velocity must be nonzero
+        for a move to succeed. Otherwise, the move fails silently.
+
+        Beckhoff EPICS motors at LCLS that have never been initialized
+        before default to a zero velocity for safety. This check
+        lets the user know why the motor does not move.
+        """
+        super().check_value(pos)
+        velo = self.velocity.get()
+        if velo <= 0:
+            raise RuntimeError(
+                f'Motor velocity is {velo}, which is not valid. '
```

Comment:
It may be wise to add the axis name in the exception message as which motor it refers to may be less clear when used outside of a GUI specifically for the axis (e.g., used in an N-D scan, in a nested function, etc).

Reply:
Good call